### ISSUE TITLE
Aurora Store crashes when clicking on an app in the store

### ISSUE THREAD

**Issue by ProPatte** [whyorean/AuroraStore#33]
**App**
Aurorastore
Version 4.2.0

**Phone**
Device: Picel 7 Pro
Operating System: GrapheneOS based on Andorid Version: 13
Granted all permissions to Aurora Store

**Description**
This happens every time:
1. Starting Aurora Store with an anonymous session
2. Clicking on an app
3. App crashes with an error

**Errormessage**

```
type: crash
osVersion: google/cheetah/cheetah:13/TQ2A.230405.003.E1/2023041100:user/release-keys
package: com.aurora.store:42
process: com.aurora.store
processUptime: 24827 + 233 ms

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.aurora.store/com.aurora.store.view.ui.details.AppDetailsActivity}: java.lang.IllegalStateException: TypeToken must be created with a type argument: new TypeToken<...>() {}; When using code shrinkers (ProGuard, R8, ...) make sure that generic signatures are preserved.
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3642)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3779)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2304)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7891)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
Caused by: java.lang.IllegalStateException: TypeToken must be created with a type argument: new TypeToken<...>() {}; When using code shrinkers (ProGuard, R8, ...) make sure that generic signatures are preserved.
	at com.google.gson.reflect.TypeToken.<init>(SourceFile:58)
	at g2.f.<init>(Unknown Source:0)
	at g2.g.a(Unknown Source:48)
	at com.aurora.store.view.ui.details.AppDetailsActivity.L0(SourceFile:373)
	at com.aurora.store.view.ui.details.AppDetailsActivity.onNewIntent(SourceFile:234)
	at com.aurora.store.view.ui.details.AppDetailsActivity.onCreate(SourceFile:860)
	at android.app.Activity.performCreate(Activity.java:8388)
	at android.app.Activity.performCreate(Activity.java:8367)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1422)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3623)
	... 14 more
```

**Comment by jerem-ma** [comment]
Same issue with my Google Pixel 2, Android 13, Lineage OS 20-20230404-NIGHTLY-walleye

**Comment by doodhout** [comment]
Reverting to version 4.1.1 seems like a valid workaround

**Comment by ProPatte** [comment]
> Reverting to version 4.1.1 seems like a valid workaround

Unfortunately not because it doesn't even load any apps on 4.1.1 for me anymore.

![Screenshot_20230416-000251](https://user-images.githubusercontent.com/18471911/232255034-ae91a06e-8dbf-4608-ad92-80d05ce0f15e.png)

**Comment by CaptainMayotoast** [comment]
I can also confirm Aurora Store crashes when selecting an application to view.  Available updates are not loading, either.  

**App**
Aurorastore
Version 4.2.0

**Phone**
Device: Pixel 7 (non-pro)
Operating System: CalyxOS 4.8.0, Android 13

**Comment by el-ef** [comment]
Same here. Aurora is broken: very slow and crashing as soon as I click on any app.

v4.2.0
Pixel 6 Pro
Graphene OS

Back to v4.1.1, works. The newer v4.2.1 does not work either, cannot log in (You are bring rate-limited).

**Comment by JimmyWalnuts** [comment]
So I'm not the only one. When starting Aurora Store (version 4.1.1) with anonymous session the app loads but doesn't populate any apps it just hangs there in a searching loop.

I first noticed this a few weeks ago when I tried to update Proton Calendar. I received error: "Failed to fetch app details" 

I tried to update Proton manually through the Aurora Store and discovered the store app isn't working.

I'm using a Pixel 6 running GrapheneOS android version 13. The phone periodically updates. Says device is up to date.

I use a VPN but I've been using it a long time, no problems until now. And it is updated.
